api/models.py

```diff
@@ -395,7 +395,7 @@ def validate_params(cls, params: dict):
     @classmethod
     def translate_fields(cls, params: dict):
         """Translate regression parameters"""
-        translated = cls.translate_fields(params)
+        translated = Node.translate_fields(params)
```

Comment:
But still, the `Node.translate_fields()` method does the same thing:
```python
    @classmethod
    def translate_fields(cls, params: dict):
        """Translate fields in `params` into objects as applicable

        Translate fields represented by strings in the `params` dictionary into
        objects that match the model.  For example, database IDs are converted
        to ObjectId.  Return a new dictionary with the translated values
        replaced.
        """
        translated = dict(cls._translate_operators(params))
        translated.update(cls._translate_object_ids(translated))
        translated.update(cls._translate_timestamps(translated))
        return translated
```
so I believe we can just drop this overridden method `Regression.translate_fields()`.  When called from Regression, it will be using the fields listed in `Revision._OBJECT_ID_FIELDS` and `Revision._TIMESTAMP_FIELDS`.  If you want to append fields to the lists provided in `Node`, then you can do that with something like this:
```python
class Regression(Node):
    _OBJECT_ID_FIELDS = Node._OBJECT_ID_FIELDS + ['regression_data.parent']
```
or by overriding it with a property which could then use `super()._OBJECT_ID_FIELDS`.

Reply:
Yes, I already got your point there but using `super()`, class variables like `_OBJECT_ID_FIELDS` was used from `Node` and not from the child class `Regression`. And that was totally as expected.
Bdw, yes we can fix it by dropping `Regression.translate_fields` and extending object ID and timestamp fields.
Thanks for the feedback.